lib/stdlib/src/json.erl

```diff
@@ -1386,16 +1386,18 @@ object_key(_, Original, Skip, Acc, Stack, Decode) ->
 
 continue(<<Rest/bits>>, Original, Skip, Acc, Stack0, Decode, Value) ->
     case Stack0 of
-        [] -> terminate(Rest, Original, Skip, Acc, Value);
+        [] -> terminate(Rest, Rest, Acc, Value);
```

Comment:
This breaks the binary pattern optimisation since the tail is now used in two paces - I'm fairly sure it will significantly affect performance

Reply:
From what we could see, looking at the beam asm, 
the only difference was that it creates a sub-binary in the case when Stack = [], 
which should be ok since that is only for the last "term" in the json string.

But I'm not fluent in beam-asm.

Reply:
It looks fine, the optimization is disabled for this particular call but not the others. Should this become a performance problem, it's not a huge deal for the compiler to allow the reused `Rest` context to be passed alongside a new tail binary representing `Rest`.

Reply:
I pushed up a new variant that postpones the creation and only does when necessary.